draftfast/optimizer.py

```diff
@@ -61,6 +86,38 @@ def solve(self):
         solution = self.solver.Solve()
         return solution == self.solver.OPTIMAL
 
+    def _set_player_constraints(self):
+        multi_constraints = dict()
+
+        for i, p in self.enumerated_players:
+            lb = 1 if p.lock else 0
+            ub = 0 if p.ban else 1
+
+            if lb > ub:
+                raise InvalidBoundsException
+
+            if p.multi_position:
+                if p.name not in multi_constraints.keys():
+                    multi_constraints[p.name] = self.solver.Constraint(lb, ub)
+                constraint = multi_constraints[p.name]
+            else:
+                constraint = self.solver.Constraint(lb, ub)
+
+            constraint.SetCoefficient(self.variables[i], 1)
+
+    def _set_player_group_constraints(self):
+        for group_constraint in self.lineup_constraints:
+            if group_constraint.exact:
+                lb = ub = group_constraint.exact
+            else:
+                lb = group_constraint.lb
+                ub = group_constraint.ub
+
+            constraint = self.solver.Constraint(lb, ub)
+            for name in group_constraint.players:
```

Comment:
The library consumer can still make this impossible by having some ridiculous group that has say three QBs and requests 2. If you do this right now, here's the error with only having constraints:

```
No solution found for command line query.
            Try adjusting your query by taking away constraints.

            Active constraints: None
            Player count: 784
```

So just to improve the messaging, I think you just need to modify `optimize` to maybe show the optimizer settings, the lineup constraints and the number of players in the pool (it also shouldn't say `command line query` which is my fault ). IMO the most frustrating thing I encounter when debugging this is why no solutions happen - this is usually user error and my hope is that this error message is helpful.

Reply:
Here's what it looks like now:
```

No solution found.
Try adjusting your query by taking away constraints.

OPTIMIZER CONSTRAINTS:

Stacks: [('NE', 100)]
Min teams: 2
No offense against D: True

LINEUP CONSTRAINTS:

Using 2 of: Philip Rivers, Sam Bradford, Andrew Luck
Using 1 to 3 of: Saints, Alshon Jeffery, Lamar Miller
Banned: Sammy Watkins, Kellen Davis
Locked: Spencer Ware

PLAYER POOL SETTINGS:

None

PLAYER COUNT: 784
```